examples/perplexity/perplexity.cpp

```diff
@@ -406,13 +404,27 @@ static results_perplexity perplexity_v2(llama_context * ctx, const gpt_params &
             const int batch_start = start + j * n_batch;
             const int batch_size  = std::min(end - batch_start, n_batch);
 
+            llama_batch batch = llama_batch_init(batch_size, 0, 1);
```

Comment:
This can be allocated and freed outside of the loop. The same batch can be re-used with `llama_batch_clear(batch)`.